tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+"""Tests for leapp upgrade of content hosts with Satellite
+
+:Requirement: leapp
+
+:CaseLevel: Integration
+
+:CaseComponent: LeappIntegration
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:CaseAutomation: Automated
+
+:Upstream: No
+"""
+import pytest
+from broker import Broker
+
+from robottelo.cli.activationkey import ActivationKey
+from robottelo.cli.repository import Repository
+from robottelo.config import settings
+from robottelo.constants import PRDS
+from robottelo.hosts import ContentHost
+from robottelo.logging import logger
+
+synced_repos = pytest.StashKey[dict]
+
+
+@pytest.fixture(scope='module')
+def module_stash(request):
+    """Module scoped stash for storing data between tests"""
+    # Please refer the documentation for more details on stash
+    # https://docs.pytest.org/en/latest/reference/reference.html#stash
+    request.node.stash[synced_repos] = {}
+    yield request.node.stash
+
+
+RHEL7_VER = '7.9'
+RHEL8_VER = '8.8'
+RHEL9_VER = '9.2'
+
+RHEL_REPOS = {
+    'rhel7_server': {
+        'id': 'rhel-7-server-rpms',
+        'name': f'Red Hat Enterprise Linux 7 Server RPMs x86_64 {RHEL7_VER}',
+        'releasever': RHEL7_VER,
+        'reposet': 'Red Hat Enterprise Linux 7 Server (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel7_server_extras': {
+        'id': 'rhel-7-server-extras-rpms',
+        'name': 'Red Hat Enterprise Linux 7 Server - Extras RPMs x86_64',
+        'releasever': '7',
+        'reposet': 'Red Hat Enterprise Linux 7 Server - Extras (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
```

Comment:
If we are eventually using the latest rhel version repositories then why setting up releasever and specific version repos? can't we use the constants or VersionedContent directly?

Reply:
We should not. 
Red Hat specifies what the [supported upgrade paths](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html-single/upgrading_from_rhel_8_to_rhel_9/index#con_supported-upgrade-paths_upgrading-from-rhel-8-to-rhel-9) are, therefore we should follow them. And that is exactly the purpose of these constants.
At this very moment, we support upgrading to the latest version of RHEL 8 which is 8.8 atm. But once 8.9 gets released, we still want to test 7.9 -> 8.8.
Once RHEL 8.10 gets released, we should manually switch our constants.

Reply:
- This has been already discussed with Lukash and Ondrej, we want specific repos to be enabled prior to get sync. 
- Constants does have any REPOS defined for specific rhel version.
- These specific rhel repos will be applicable for leapp scenarios so defined in same test module.

Reply:
@Gauravtalreja1 Can we resolve this one?

Reply:
Sure, its non-blocking